libz-rs-sys/src/gz.rs

```diff
@@ -296,72 +296,49 @@ unsafe fn gzopen_help(source: Source, mode: *const c_char) -> gzFile {
 }
 
 // Format a fake file path corresponding to an fd, for use in error messages.
-//
-// # Returns
-//
-// * A pointer to a null-terminated C string that the caller now owns.
-// * Or a null pointer if an error occurs.
-//
-// # Safety
-//
-// The return value, if non-null, must be freed using only `ALLOCATOR`.
-unsafe fn fd_path(fd: c_int) -> *const c_char {
-    // This is equivalent to `format!("<fd:{}>\0", fd)`, but without the dependency on std::format.
+fn fd_path(buf: &mut [u8; 17], fd: c_int) -> &CStr {
```

Comment:
The `[u8; 17]` could use a comment to document why it's that size. (6 bytes for `"<fd:>\0"`, 11 bytes for the decimal representation of the smallest possible `i32` value, including the minus sign.)

It's probably also worth documenting that this function will truncate the formatted string (on the left) in the extremely unlikely event that the buffer is too small -- which could happen on an ILP64 system where `c_int` is `i64`, if this code is used in a process that has more than 1e10 open file descriptors.

Reply:
I've just bumped the buffer size to be sure we never run into issues